yt_dlp/extractor/cda.py

```diff
@@ -83,8 +98,92 @@ def _download_age_confirm_page(self, url, video_id, *args, **kwargs):
                 'Content-Type': content_type,
             }, **kwargs)
 
+    def _login(self, username, password):
```

Comment:
rename this to `_perform_login` and remove it from real_extract

Reply:
how should I fetch the token then?

Reply:
`_perform_login` is automatically called during `_real_initialize`

Reply:
but how is the extractor supposed to get the token? it was being returned by the function. now I reworked it to get it out of cache but that feels like a hack too